src/nfa/nfa.cc

```diff
@@ -330,15 +330,21 @@ void OnTheFlyAlphabet::add_symbols_from(const StringToSymbolMap& new_symbol_map)
     }
 }
 
-///// Nfa structure related methods
+size_t Delta::size() const
+{
+    size_t res = 0;
+    for (const auto& trans : *this) res++;
+
+    return res;
+}
 
-void Nfa::add_trans(State state_from, Symbol symbol, State state_to) {
-    // TODO: Define own exception.
-    if (!is_state(state_from) || !is_state(state_to)) {
-        throw std::out_of_range(std::to_string(state_from) + " or " + std::to_string(state_to) + " is not a state.");
+void Delta::add(State state_from, Symbol symbol, State state_to)
+{
+    if (state_from >= post.size()) {
```

Comment:
little thing, maybe using push_back instead of resize might be more robust wrt change of a data structure (like vector to deque)

Reply:
you mean applying push_back `state_from - post.size()` times. Because I think we don't have guaranteed that `state_from` is bigger just by one.